storage/disk/txn.go

```diff
@@ -318,11 +318,29 @@ func (txn *transaction) partitionWrite(op storage.PatchOp, path storage.Path, va
 }
 
 func (txn *transaction) partitionWriteMultiple(node *partitionTrie, path storage.Path, value interface{}, result []update) ([]update, error) {
-
 	// NOTE(tsandall): value must be an object so that it can be partitioned; in
 	// the future, arrays could be supported but that requires investigation.
-	obj, ok := value.(map[string]interface{})
-	if !ok {
+
+	switch v := value.(type) {
+	case map[string]interface{}:
+		bs, err := serialize(v)
+		if err != nil {
+			return nil, err
+		}
+		return txn.doPartitionWriteMultiple(node, path, bs, result)
+	case json.RawMessage:
+		return txn.doPartitionWriteMultiple(node, path, v, result)
+	case []uint8:
+		return txn.doPartitionWriteMultiple(node, path, v, result)
+	}
+
+	return nil, &storage.Error{Code: storage.InvalidPatchErr, Message: "value cannot be partitioned"}
+}
+
+func (txn *transaction) doPartitionWriteMultiple(node *partitionTrie, path storage.Path, bs []byte, result []update) ([]update, error) {
+	var obj map[string]json.RawMessage
```

Comment:
If we're passing in a `json.RawMessage` that happens to not be an object (like any other json value, e.g. `"string"`), this would probably break. Do we know, up the call stack, that this cannot happen?

Reply:
In the `partitionWriteMultiple` method which calls `doPartitionWriteMultiple`, we expect `value` to be an object. If  `json.RawMessage` is not an object is passed to `doPartitionWriteMultiple` it will generate an error with message `value cannot be partitioned` which is what we would expect, correct ?